lib/plugin.js

```diff
@@ -4,33 +4,52 @@ const _ = require('lodash')
 const routes = require('./routes')()
 const adapterUtils = require('./utils/adapter')()
 const routeUtils = require('./utils/route')()
+const includes = require('./includes');
+
+// constants
+const MIME = {
+    standard: ['application/vnd.api+json', 'application/json'],
+    patch: ['application/json-patch+json']
+};
 
 exports.register = function (server, opts, next) {
     server.expose('version', require('../package.json').version);
     
     const adapter = opts.adapter;
-    
+    const schemas = {};
+
     adapterUtils.checkValidAdapter(adapter);
     
     adapter.connect(() => {
         server.expose('adapter', adapter);
         next()
     });
-    
+
+    const appendLinkedResources = includes(adapter, schemas).appendLinkedResources;
+
+    function onRouteRegister(schema) {
```

Comment:
it is, everything ES5 is valid ES6. But it'd be nice if we made use of new features.